include/sir/helpers.h

```diff
@@ -436,7 +436,7 @@ bool _sir_getchar(char* input);
 # define _sir_snprintf_trunc(dst, size, ...) \
     do { \
       volatile size_t _n = size; \
-      if (!snprintf(dst, _n, __VA_ARGS__)) { (void)_n; }; \
+      if (!snprintf(dst, (short)_n, __VA_ARGS__)) { (void)_n; }; \
```

Comment:
we will get compiler warnings when using the standard snprintf if we do this i'm pretty sure. the portable one is `size_t` which isn't always the same type, in fact, it's almost never the same. It could be unsigned etc. You might want to do an #if defined() here, or at least test if casting to short in master causes a warning in CI

Reply:
It shouldn't ever cause a warning because the cast is explicit.  CI here has already these this - the DPSPRINTF isn't enabled yet on any of the build stage builds.  However, it will cause an warning if the mismatch is the other way, so technically, I should really cast it to a char =)  Did that in c5acfa0

Reply:
Actually this function is getting gross, I'll try to solve this a saner way.